core/arch/arm/kernel/elf_load.c

```diff
@@ -372,14 +611,34 @@ TEE_Result elf_load_body(struct elf_load_state *state, vaddr_t vabase)
 		return res;
 
 	if (state->shdr) {
-		/* Process relocation */
-		for (n = 0; n < ehdr->e_shnum; n++) {
-			if (state->shdr[n].sh_type == SHT_RELA)
-				return TEE_ERROR_NOT_IMPLEMENTED;
-			else if (state->shdr[n].sh_type == SHT_REL) {
-				res = elf_process_rel(state, n, vabase);
-				if (res != TEE_SUCCESS)
-					return res;
+		if (state->is_32bit) {
+			Elf32_Ehdr *ehdr = state->ehdr;
+			Elf32_Shdr *shdr = state->shdr;
+
+			/* Process relocation */
+			for (n = 0; n < ehdr->e_shnum; n++) {
+				if (shdr[n].sh_type == SHT_RELA)
+					return TEE_ERROR_NOT_IMPLEMENTED;
+				else if (shdr[n].sh_type == SHT_REL) {
+					res = e32_process_rel(state, n, vabase);
+					if (res != TEE_SUCCESS)
+						return res;
+				}
+			}
+		} else {
+			Elf64_Ehdr *ehdr = state->ehdr;
+			Elf64_Shdr *shdr = state->shdr;
+
+			/* Process relocation */
+			for (n = 0; n < ehdr->e_shnum; n++) {
+				if (shdr[n].sh_type == SHT_RELA) {
+					res = e64_process_rela(state, n,
```

Comment:
same remark as above on the macro, adding the change in the function exx_process_rela